chainstate/src/detail/chainstateref/mod.rs

```diff
@@ -185,6 +193,31 @@ impl<'a, S: BlockchainStorageRead, V: TransactionVerificationStrategy> Chainstat
             .map_err(PropertyQueryError::from)
     }
 
+    pub fn get_existing_gen_block_index(
```

Comment:
Why is this function necessary? Why not use the one above it?

Reply:
By "the one above it" do you mean `get_gen_block_index`? The "get_existing" variant implies that the block index must exist and returns an error when it doesn't, while `get_gen_block_index` returns an Option. It is not necessary but it reduces the verbosity of the caller code.
If you mean `get_existing_block_index` (i.e. the one without "gen") - well sometimes it's more convenient to work with "gen" variants of Ids and Indices, sometimes it's not. So, 2 variants of each function is often desired.

Btw, `get_existing_gen_block_index` is currently unused, I only used it in an intermediate implementation. So, I can remove it or keep it "for completeness".

Reply:
Yes, please remove it. I don't think such redundancy is a good idea.